### PR TITLE
Check the removeType in IsDropCitusStmt

### LINKED ISSUE #3856: IsDropCitusStmt possible bug
IsDropCitusStmt checks if the name is 'citus' but it doesn't check the `removeType` of `DropStmt`. 

We have the following code in utility hook:

```c
	if (IsDropCitusStmt(parsetree))
	{
		StopMaintenanceDaemon(MyDatabaseId);
	}
```

So if we are dropping a table named 'citus' it might possibly stop the maintenance daemon I guess even though it shouldn't.

### PR DESCRIPTION
**Pull request by SaitTalhaNisanci**
We should check the remove type in IsDropCitusStmt because if the remove
type is not OBJECT_EXTENSION then the stored objects in
dropStmt->objects may not be of type Value. This was crashing PG-13.

Fixes #3856.

### CODE REVIEW

File: src/backend/distributed/commands/extension.c [hunk 1]
```diff
@@ -729,10 +729,15 @@ IsDropCitusStmt(Node *parseTree)
 	{
 		return false;
 	}
+	DropStmt *dropStmt = (DropStmt *) parseTree;
```
Comment (onurctirtir) [inline src/backend/distributed/commands/extension.c:732]: minor:

```suggestion

    /* check if the drop command is a DROP EXTENSION command */
	DropStmt *dropStmt = (DropStmt *) parseTree;
```

Comment (SaitTalhaNisanci) [reply src/backend/distributed/commands/extension.c:732]: Actually we can maybe rename the method as `IsDropCitusExtensionStmt`, what do you think?

Comment (onurctirtir) [reply src/backend/distributed/commands/extension.c:732]: `IsDropCitusExtensionStmt` would be a really better function name

Comment (SaitTalhaNisanci) [reply src/backend/distributed/commands/extension.c:732]: @onurctirtir, updated some docs as well, does it look good now?

Comment (onurctirtir) [reply src/backend/distributed/commands/extension.c:732]: yes

**Pull request merged**